mypy/semanal.py

```diff
@@ -2983,12 +2993,16 @@ def infer_reachability_of_if_statement(s: IfStmt,
                                        platform: str) -> None:
     for i in range(len(s.expr)):
         result = infer_if_condition_value(s.expr[i], pyversion, platform)
-        if result == ALWAYS_FALSE:
+        if result in (ALWAYS_FALSE, MYPY_FALSE):
             # The condition is always false, so we skip the if/elif body.
             mark_block_unreachable(s.body[i])
```

Comment:
Just to make sure, the intention is that `if not TYPE_CHECKING:` means "don't type check"? That's fine but not explicitly documented in PEP-484.

In this case
```Python
if TYPE_CHECKING:
    import expensive_mod
else:
    # I don't care about runtime annotations
    expensive_mod = CleverPseudoModule('SomeClass')

def a_func(arg: expensive_mod.SomeClass) -> None:
    ...
```
The else branch will not be checked at all. Again, it's fine, but I think it should be documented.